coupled_laplace_problem/coupled_laplace_problem.cc

```diff
@@ -0,0 +1,640 @@
+  const double max_delta_t = precice.advance(computed_timestep_length);
+
+  // As a next step, we obtain data, i.e. the boundary condition, from another
+  // participant. We have already all IDs and just need to convert our obtained
+  // data to the deal.II compatible 'boundary map' , which is done in the
+  // format_deal_to_precice function.
+  precice.readBlockScalarData(read_data_id,
+                              n_interface_nodes,
+                              interface_nodes_ids.data(),
+                              read_data.data());
+
+  format_precice_to_deal(boundary_data);
+
+  return max_delta_t;
+}
+
+
+
+// This function takes the std::vector obtained by preCICE in `read_data` and
+// inserts the values to the right position in the boundary map used throughout
+// our deal.II solver for Dirichlet boundary conditions. The function is only
+// used internally in the Adapter class and not called in the solver itself. The
+// order, in which preCICE sorts the data in the `read_data` vector is exactly
+// the same as the order of the initially passed vertices coordinates.
+template <int dim, typename ParameterClass>
+void
+Adapter<dim, ParameterClass>::format_precice_to_deal(
+  std::map<types::global_dof_index, double> &boundary_data) const
+{
+  // We stored already the coupling DoF indices in the `boundary_data` map, so
+  // that we can simply iterate over all keys in the map.
+  auto dof_component = boundary_data.begin();
+  for (int i = 0; i < n_interface_nodes; ++i)
+    {
+      AssertIndexRange(i, read_data.size());
+      boundary_data[dof_component->first] = read_data[i];
+      ++dof_component;
+    }
+}
+
+
+// The solver class is essentially the same as in step-4. We only extend the
+// stationary problem to a time-dependent problem and introduced the coupling.
+// Comments are added at any point, where the workflow is differs from step-4.
+template <int dim>
+class CoupledLaplaceProblem
+{
+public:
+  CoupledLaplaceProblem();
+
+  void
+  run();
+
+private:
+  void
+  make_grid();
+  void
+  setup_system();
+  void
+  assemble_system();
+  void
+  solve();
+  void
+  output_results() const;
+
+  Triangulation<dim> triangulation;
+  FE_Q<dim>          fe;
+  DoFHandler<dim>    dof_handler;
+
+  SparsityPattern      sparsity_pattern;
+  SparseMatrix<double> system_matrix;
+
+  Vector<double> solution;
+  Vector<double> old_solution;
+  Vector<double> system_rhs;
+
+  // Here, we allocate all structures required for the preCICE coupling. The map
+  // is used to apply Dirichlet boundary conditions and filled in the Adapter
+  // class with data from our other participant. The CouplingParameters hold the
+  // preCICE configuration as described above. The interdace boundary ID is the
+  // ID associated to our coupling interface and needs to be specified, when we
+  // set up the Adapter class object, because we pass it directly to the
+  // Constructor of this class.
```

Comment:
```suggestion
  // We allocate all structures required for the preCICE coupling. The map
  // is used to apply Dirichlet boundary conditions and filled in the Adapter
  // class with data from the other participant. The CouplingParameters hold the
  // preCICE configuration as described above. The interdace boundary ID is the
  // ID associated to our coupling interface and needs to be specified, when we
  // set up the Adapter class object, because we pass it directly to the
  // Constructor of this class.
```